### ISSUE TITLE
Specify credentials & db directory & db creating

### ISSUE THREAD

**Issue by stepnoy** [couchbaselabs/couchbase-lite-local#1]
There is no way to use fixed credential every time. It create new random credentials every time when I run it.

And it is not clear for me where db files will be created to hold data. It would be great to have an option to specify place where file should be.

And one more thing. Right now I couldn't create db by rest api. It returns me an error.

**Comment by sideshowcoder** [comment]
Hi thanks for trying it out, the database files will be created in the directory you run the jar in, under data. The reason for it not being an option right now is that as Couchbase lite is primarily developed for mobile it does not expose a nice interface to do so right now, but there is a way todo it so I will see to make it work.

About the random credentials, the idea is that you would start this as part of your application and then use the credentials it provides but I see the point that this is making things harder than it should be so thanks for the suggestion! :+1: 

About the creating a DB do you mind posting the error and the curl command you run so I can look into it? 

Thanks

**Comment by stepnoy** [comment]
There it is

 curl --user 132fca6f-385a-4742-8a8e-4e08405817ff:4dda8c83-4b02-4f7e-a2b9-dd24e7118d7f  -X PUT    http://localhost:5984/testbase 

{"error":"not_found","reason":"Router unable to route request to do_PUT_Databasejava.lang.reflect.InvocationTargetException"}

When I am trying to create base (with PUT request) server catches an exception. 

Sep 3, 2014 4:31:59 PM com.couchbase.lite.util.SystemLogger e
SEVERE: Router: Router unable to route request to do_PUT_Database
java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at com.couchbase.lite.router.Router.start(Router.java:464)
    at com.couchbase.lite.listener.LiteServlet.service(LiteServlet.java:128)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
    at Acme.Serve.Serve$ServeConnection.runServlet(Serve.java:2251)
    at Acme.Serve.Serve$ServeConnection.parseRequest(Serve.java:2176)
    at Acme.Serve.Serve$ServeConnection.run(Serve.java:1988)
    at Acme.Utils$ThreadPool$PooledThread.run(Utils.java:1223)
    at java.lang.Thread.run(Thread.java:695)
Caused by: java.lang.UnsupportedClassVersionError: com/couchbase/lite/storage/JavaSQLiteStorageEngine : Unsupported major.minor version 51.0
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClassCond(ClassLoader.java:637)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:621)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:471)
    at org.jruby.util.JRubyClassLoader.findClass(JRubyClassLoader.java:158)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:171)
    at com.couchbase.lite.storage.SQLiteStorageEngineFactory.createStorageEngine(SQLiteStorageEngineFactory.java:41)
    at com.couchbase.lite.Database.open(Database.java:906)
    at com.couchbase.lite.router.Router.do_PUT_Database(Router.java:731)
    ... 12 more

**Comment by sideshowcoder** [comment]
Thanks I will look into that, I tried it out locally but couldn't reproduce it right away sorry :(.

Are you running the Jar which is downloaded or did you build it yourself? It looks to me like there is a missmatch with SQLite 

```
JavaSQLiteStorageEngine : Unsupported major.minor version 51.0
```

I only ran it on my Pi running Raspbian so far so maybe thats the problem.

**Comment by stepnoy** [comment]
I'll try it on my PI and send you results.

**Comment by stepnoy** [comment]
I have tested lib on my Pi. Looks like it works great at Pi. This error is only appearing when run on mac.

**Comment by sideshowcoder** [comment]
@stepnoy thats a good start I guess, so now we need to figure out what is going wrong on the Mac. Could you provide information what you are running there?

**Comment by sideshowcoder** [comment]
Any news on this? I uploaded a new version of the Jar and I'm unable to reproduce so I would close this if you run intro trouble again let me know please. I also implemented passing the credentials in the version on master and am working on specifying the directories for the files see #2 and #3 for more details

**Comment by achyuthrbt** [comment]
Does the above issue was fixed ? Because I'm getting the same error as mentioned above when I try to RUN the jar file in my Ubuntu Machine.

I didn't build the module on my own, I just downloaded the jar from the following link : [https://github.com/couchbaselabs/couchbase-lite-local/releases](url) - (In this version 0.0.3 and 0.0.4) in both the Jar's I am getting the same error.

Any help will be appreciated.

Thanks in Advance,
Achu
